src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+import java.util.NoSuchElementException;
+
+public class ArrayList<E> implements List<E> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private E[] data;
+    private int size;
 
+    public ArrayList() {
+        data = (E[]) new Object[DEFAULT_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public void add(E value) {
+        if (isStorageFull()) {
+            grow();
+        }
+        data[size] = value;
+        size++;
     }
 
     @Override
-    public void addAll(List<T> list) {
-
+    public void add(E value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add element: " + value
+                    + " on the position: " + index);
+        } else if (isStorageFull()) {
```

Comment:
we don't need else after throwing an exception
```suggestion
        } 
        if (isStorageFull()) {
```

Reply:
Done!